src/ocm/services/ClusterDetailsService.ts

```diff
@@ -80,7 +81,12 @@ const ClusterDetailsService = {
       cluster,
       ...args,
     });
-    const cpuArchitecture = cluster?.cpuArchitecture || CpuArchitecture.x86;
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const { search } = useLocation();
```

Comment:
@celdrake can this hook be used outside a component?

Reply:
https://reactjs.org/docs/hooks-rules.html

Reply:
Agree, ideally we should not break the hook rules.
This can be fixed by leaving the `useLocation` as it was in the previous commit, and passing the `defaultArchitecture` to this component. So the logic for the parameter is outside the component.